docs/apm/api.asciidoc

```diff
@@ -708,3 +709,87 @@ curl -X DELETE "http://localhost:5601/api/apm/sourcemaps/apm:foo-1.0.0-644fd5a9"
 --------------------------------------------------
 {}
 --------------------------------------------------
+
+////
+*******************************************************
+*******************************************************
+////
+
+[role="xpack"]
+[[agent-key-api]]
+=== Agent Key API
+
+The Agent Key API allows you to configure agent keys to authorize requests from agents to the APM Server.
+
+The following Agent key APIs are available:
+
+* <<apm-create-agent-key>> to create an agent key
+
+[float]
+[[use-agent-key-api]]
+==== How to use APM APIs
+
+.Expand for required headers, privileges, and usage details
+[%collapsible%closed]
+======
+include::api.asciidoc[tag=using-the-APIs]
+======
+
+////
+*******************************************************
+////
+
+[[apm-create-agent-key]]
+==== Create agent key
+
+[[apm-create-agent-key-req]]
+===== Request
+
+`POST /api/apm/agent_keys`
+
+[role="child_attributes"]
+[[apm-create-agent-key-req-body]]
+===== Request body
+
+`name`::
+(required, string) Name of the agent key.
+
+`event`::
+(optional, boolean) Adds the `event:write` privilege. Required for ingesting agent events.
+
+`agentConfig`::
+(optional, boolean) Adds the `config_agent:read` privilege. Required for agents to read agent configuration remotely.
+
+`sourcemap`::
+(optional, boolean) Adds the `sourcemap:write` privilege. Required for uploading sourcemaps.
+
+[[apm-agent-key-create-example]]
+===== Example
+
+[source,curl]
+--------------------------------------------------
+POST /api/apm/agent_keys
+{
+    "name": "apm-key",
+    "event": true,
+    "agentConfig": false,
+    "sourcemap": false
```

Comment:
I'm a little late to the party but I think it would be better to stick with the original privilege names. 

```suggestion
    "privileges": ["event:write", "config_agent:read", "sourcemap:write"]
```

The new ones are barely shorter, and they just introduce new terminology. Also, I think `event:write` is more clear that it provides write access, whereas `event: true` is more opaque

Reply:
Changed privileges to an array in the code and docs